Compiler/src/typelattice.jl

```diff
@@ -799,3 +741,13 @@ function stoverwrite1!(state::VarTable, change::StateUpdate)
     state[changeid] = newtype
     return state
 end
+
+# The ::AbstractLattice argument is unused and simply serves to disambiguate
+# different instances of the compiler that may share the `Core.PartialStruct`
+# type.
+function Core.PartialStruct(::AbstractLattice, @nospecialize(typ), fields::Vector{Any})
+    for i = 1:length(fields)
+        assert_nested_slotwrapper(fields[i])
+    end
+    return Core._PartialStruct(typ, fields)
+end
```

Comment:
What’s the reason for performing this disambiguation specifically for `PartialStruct`? Would similar changes be needed for other `Core`-lattice types as well?

Reply:
`assert_nested_slotwrapper` references Compiler-internal lattice elements, so it needs to be specialized if you load a second copy of the compiler.